controller/api_0.0.0/video.js

```diff
@@ -0,0 +1,62 @@
+let router          = require('express').Router();
+let op              = require('../../model/api_operations');
+let getVideoInfo    = require('../function/video_info');
+
+router.post(
+    '/api/0.0.0/submit_video',
+    require('../../middleware/logged_in_only.js'),
+    (req, res) => {
+        // verify all the data come thru
+        // need to have the url, by, and channel
+
+        let url = req.body.url;
+        let who_submit = req.body.who_submit;
```

Comment:
User does not know what their ID is, they only know their username. I'd suggest you use the `req.session.passport.user.id` instead -which is set by our authentication system and would be available for all logged in user.

Reply:
When front-end is implemented, we can look up current page's [`window.location.href`](https://developer.mozilla.org/en-US/docs/Web/API/Window/location) and automatically include it as a sort of [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden) input in the form. So it would come from user, but user wouldn't have to lookup or manually enter anything.